### PR TITLE
feat!: Support YQ dialect

### PR DESCRIPTION
**Pull request by Raubzeug**
Closes #ISSUE_ID

### CODE REVIEW

File: src/autocomplete/databases/yql/grammar/YQL.g4 [hunk 1]
```diff
@@ -333,25 +337,25 @@ expr_list
     ;
 
 pure_column_list
-    : LPAREN an_id (COMMA an_id)* RPAREN
+    : LPAREN an_existed_column_id (COMMA an_existed_column_id)* RPAREN
```
Comment (roberthovsepyan) [inline src/autocomplete/databases/yql/grammar/YQL.g4:368]: `existed` is not grammatically correct here, `existing` is the word you're looking for

File: src/autocomplete/autocomplete-types.ts [hunk 2]
```diff
@@ -154,6 +155,7 @@ export interface AutocompleteData<
     ignoredTokens: Set<number>;
     rulesToVisit: Set<number>;
     enrichAutocompleteResult: EnrichAutocompleteResult<A>;
+    context?: ParserRuleContext;
```
Comment (roberthovsepyan) [inline src/autocomplete/autocomplete-types.ts:158]: why do we need this? can't see it being used anywhere

File: src/autocomplete/databases/yql/tests/empty-query-yq.test.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import {KeywordSuggestion} from '../../../autocomplete-types';
```
Comment (roberthovsepyan) [inline src/autocomplete/databases/yql/tests/empty-query-yq.test.ts:1]: nit: I'd move yq tests into a separate folder

**Review by roberthovsepyan — CHANGES_REQUESTED** [review]

File: src/autocomplete/autocomplete-types.ts (hunk 2 above)
Comment (Raubzeug) [reply src/autocomplete/autocomplete-types.ts:158]: > why do we need this? can't see it being used anywhere

We use context here: https://github.com/gravity-ui/websql-autocomplete/pull/173/files#diff-038083b7de04219ec7c935fd3387dbc39cb1b97b34e468a0539d4cdd2f58bbc6R80 . It is needed to arrange an entrypoint for parsing query to suggest keywords. If no context is given, all tree will be taken, no matter what you use in `parseTree` function.

File: src/autocomplete/shared/parse-query-with-cursor.ts [hunk 4]
```diff
@@ -33,6 +34,10 @@ export function parseYQLQueryWithCursor(queryWithCursor: string): YQLAutocomplet
     return parseYQLQuery(...separateQueryAndCursor(queryWithCursor));
 }
 
+export function parseYQQueryWithCursor(queryWithCursor: string): YQLAutocompleteResult {
+    return parseYQQuery(...separateQueryAndCursor(queryWithCursor));
```
Comment (NikitaShkaruba) [inline src/autocomplete/shared/parse-query-with-cursor.ts:38]: We don't have abbreviations in all caps in our project. So please in the next PR, refactor your functions to `parseYqQueryWithCursor`, `parseYqlQueryWithCursor`.

**Review by NikitaShkaruba — APPROVED** [review]

**Review by roberthovsepyan — APPROVED** [review]

**Pull request merged**